common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -999,13 +1000,22 @@ public static UTF8String concatWs(UTF8String separator, UTF8String... inputs) {
   }
 
   public UTF8String[] split(UTF8String pattern, int limit) {
+    return split(pattern, limit, false);
+  }
+
+  public UTF8String[] split(UTF8String pattern, int limit, boolean ifQuote) {
```

Comment:
I think internal consistency is more important than being compatible with other systems. It's super weird that `split` and `split_part` functions interpret the delimiter parameter differently.

Reply:
cc @srielau

Reply:
Sure, this will be a very important decision and will change implementation details. We should make a call before continuing checking other code details.

Current split treats `pattern` as a regex, while split_part treats `pattern` as a fixed pattern. Because we know that split_part can be turned into ElementAt(split()), that is why I introduced branches into existing code path: to re-use code as much as possible while maintaining minor differences between function specs.

There are two principles:
1. Implement a function that is aligned with most of the vendors.
2. Re-use code as much as possible.

Thus leads to two options:
Option 1: we implement split_part separately without re-using element_at and split, this will make the behavior compatible with others but might not produce minimal code addition.
Option 2: we change split_part to follow split,  thus leads to very nice code re-use, but our split will be pretty unique.

What do you think @cloud-fan @srielau?

Reply:
In another discussion, I believe it is encouraged to copy code and do modification (thus option 1): https://github.com/apache/spark/pull/35534/files#r810761848?

Reply:
> but our split will be pretty unique.

Is it already the case for the existing `split` function?

Reply:
Let's go back to first principles: Why are we adding split_part? Who is expected to use it?
Is the need organic, because users want a faster way to write something they can already do?
Or is the need to fill a functional gap and there are expectations of a different behavior?

Reply:
OK makes sense. Shall we also add a variant of `split` function that uses the delimiter semantic of `split_part`? This seems like a missing piece of the split functionalities in Spark.

Reply:
Sounds good. I will change to adding a variant of `split` that is aligned with delimiter semantic of `split_part`

Reply:
I tried to add a `split_delimiter`. Please take a look. I am happy to update `split_delimiter` if you think the code can be further unified somehow.